docs/cookbooks/sharepoint.md

```diff
@@ -0,0 +1,114 @@
+
+Essa string precisa estar disponível como uma variável de ambiente `MS365_KEY` no seu ambiente de desenvolvimento local para descriptografar o token e no Github Actions para que o token armazenado no arquivo `ms365.rds` possa ser descriptografado.
+
+Como o token fica exposto no Github ele deve ser criptografado com a chave gerada e será descriptografado posteriormente para autenticação no sharepoint durante a execução do Actions. Dessa forma o processo pode ocorrer no Github actions sem expor o token de autenticação publicamente.
+
+
+
+## Recuperação do Token criptografado
+
+Após o token ser disponibilizado como um arquivo .rds criptografado, para utilizá-lo execute o seguinte script R:
+
+```R
+# decriptar token criptografado
+k_env <- Sys.getenv("MS365_KEY") # obtém a string da chave privada que foi armazenada como variável de ambiente
+key <- cyphr::key_sodium(sodium::hex2bin(k_str)) # converte a chave privada de string para binário
+token = cyphr::decrypt(readRDS("ms365r.rds"), key) # le e descriptografa o arquivo do token
+
+```
+
+Dessa forma o token pode ser utilizado para autenticação e obter os arquivos 
+
+
+
+## Disponibilização da chave privada MS36_KEY para o Github Actions
+
+No nosso plano atual do Github para a organização `splor-mg` "organization secrets and variables cannot be used by private repositories".
+Portanto é necessário criar o segredo `MS365_KEY` em cada repositório que precisa de realizar extração.
+
+Nas configurações do repositório (Settings), selecione "Secrets and Variables" e "actions":
+
+![](sharepoint_img/github_actions_secret_1.png)
+
+Insira um novo segredo `MS36_KEY`  e a string da chave privada anotada:
+
+![](sharepoint_img/github_actions_secret_2.png)
+
+Nota: Se você gerar outra chave privada e criptografar o arquivo com essa nova chave, naturalmente a chave anterior não funcionará para descriptografar o arquivo do token.
+
+Importante lembrar que a variável de ambiente `MS365_KEY` deve estar disponível dentro do container, no `Makefile` do projeto:
+
+
+```makefile
+env:
+	MS3655_KEY: ${{ secrets.MS365_KEY }}
+run: 
+	docker run -e MS365_KEY=$MS365_KEY --rm --mount type=bind,source=${PWD},target=/project fjuniorr/${{ github.event.repository.name }}
+```
+
+
+
+## Utilização do Token para acesso ao Sharepoint
+
+Após todo o processo descrito até aqui, o token pode ser utilizado para autenticação e acesso a recursos no sharepoint, como no script R exemplo a seguir:
+
+```R
+library(Microsoft365R)
+library(cyphr)
+
+k_env <- Sys.getenv("MS365_KEY") # Chave privada obtido via secret vindo do Gitgub Actions
+
+# decriptar token
+key <- cyphr::key_sodium(sodium::hex2bin(k_str))
```

Comment:
@labanca seguindo o tutorial, em um script do R eu fiz os passos de geração do arquivo `ms365r.rds` em um repositório em minha máquina e salvei a variável de ambiente `MS365_KEY`. 

No script de extração de dados do `sharepoint/onedrive` eu inseri as linhas de código desta seção:
```R
library(Microsoft365R)
library(frictionless)
library(cyphr)

k_env <- Sys.getenv("MS365_KEY")
key <- cyphr::key_sodium(sodium::hex2bin(k_str))
token = cyphr::decrypt(readRDS("ms365r.rds"), key)

extract <- function() {
  package <- suppressMessages(read_package("datapackage.yaml"))
  source <- package$sources[[1]]
  site <- get_sharepoint_site(site_url = source[["path"]])
  docs <- site$get_drive(source[["drive"]])
  folder <- docs$get_item(source[["item"]])
  folder$download(source[["dest"]], overwrite = TRUE)
}

extract()

```

 e retornou a seguinte mensagem:
`Error: object 'k_str' not found`

Reply:
Fiz o push corrigindo @hslinhares. Esqueci de mudar o nome das variáveis k_str pra ficar k_env igual na linha de cima.

Veja se corrigiu.

Reply:
@labanca corrigiu sim